rllib/env/single_agent_env_runner.py

```diff
@@ -447,7 +447,7 @@ def _sample_episodes(
         obs, infos = self.env.reset()
         for env_index in range(self.num_envs):
             episodes[env_index].add_env_reset(
-                observation=obs[env_index],
+                observation=unbatch(obs)[env_index],
```

Comment:
Strange: Shouldn't this already have caused a bug in the existing flatten obs example (which also uses a dict obs space)?

Reply:
I think, the flatten obs example silently snuck around it: The `unbatch(obs)`  was only forgotten in `_sample_episodes` and not in `_sample_timesteps` (has cost me a lot of time to figure it out thoug). Because the flatten obs does no evaluation - it never ran into it :D

Reply:
Ahhh, yes, makes perfect sense. Thanks for catching this!

Reply:
With `ray==2.24.0`, `batch_mode="complete_episodes"` will cause the bug. And `obs = unbatch(obs)` should outter the for-loop like what `_sample_timesteps` do:
```python
        obs, infos = self.env.reset()
        obs = unbatch(obs)
        for env_index in range(self.num_envs):
            episodes[env_index].add_env_reset(
                observation=obs[env_index],
```